deref_schema.js

```diff
@@ -18,7 +18,7 @@ function readJsonSchema(filePath) {
 function filenameMatches(filename, allowedNames, allowedExt) {
     for (const name of allowedNames) {
         const allowedOption = name + allowedExt;
-        if (filename.toLowerCase() == allowedOption.toLowerCase()) {
+        if (filename.toLowerCase() === allowedOption.toLowerCase()) {
```

Comment:
just out of curiosity : why does the type check(===) here is needed ?

Reply:
It makes no functional difference here, but using `===`/`!==` instead of `==`/`!=` is a good habit and I intend to make it a lint rule at some point in the future. At the moment, this change was just to silence a warning from WebStorm.

The reason it's a good habit is because `==` has [very loose rules](https://dorey.github.io/JavaScript-Equality-Table/), and it's never obvious if program code relies on that looseness. You end up with code that's easy to make bugs in.

There is one exception to this rule: `!= null` is identical to `!== null && !== undefined`, which is a really common test - often you don't know whether incoming data uses `null`s or `undefined`s when something is missing, so it's best to catch both unless you have a good reason to be specific. `== null` is also fine for the same reason.

Reply:
Thanks @LachlanStuart  for very explanatory answer! The web site with the table of == and === usage is a great source as well!